### ISSUE TITLE
SignatureChecker ignores annotation of enclosing class when checking field type

### ISSUE THREAD

**Issue by Marcono1234** [mojohaus/animal-sniffer#131]
It looks like `SignatureChecker` ignores whether the enclosing class is annotated (and errors should be ignored) when processing field types:
https://github.com/mojohaus/animal-sniffer/blob/364fc20773cfdde514a212a683c344d3a253bb8b/animal-sniffer/src/main/java/org/codehaus/mojo/animal_sniffer/SignatureChecker.java#L362

This should probably be `checkType(Type.getType(descriptor), ignoreError);`.

**Comment by cpovirk** [comment]
I think I hit this, as well. I attempted to work around it by putting `@IgnoreJRERequirement` on the field itself, but that turns out not to be a valid target:
https://github.com/mojohaus/animal-sniffer/blob/8c2eac0f72792c6225f45125f9893db47da4f439/animal-sniffer-annotations/src/main/java/org/codehaus/mojo/animal_sniffer/IgnoreJRERequirement.java#L39

(At first, I was using our own custom `@IgnoreJRERequirement` annotation ([background](https://github.com/google/guava/pull/3652)). That annotation is defined to allow us to place it on fields, but presumably Animal Sniffer doesn't look there for it, since it's not normally valid.)

**Comment by Marcono1234** [comment]
This is not fixed in 1.22; the comment on PR #139 erroneously referred to this issue (if I see that correctly). This can still be seen with the following simple example (using Java 5 signatures):
```java
import java.util.Optional;
import org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement;

@IgnoreJRERequirement
class MyClass {
    // BUG: This is not ignored
    Optional<String> f;
    
    // This is ignored due to @IgnoreJRERequirement, as desired
    Optional<String> test() {
        return null;
    }
}
```

**Comment by TimvdLippe** [comment]
Can confirm that this issue has not been fixed yet: https://github.com/mockito/mockito/pull/2735
